src/rgw/support/rgw_support_bundle

```diff
@@ -117,32 +133,95 @@ class RGWSupportBundle:
             # include the latest 5 log files of motr traces in support bundle
-            list_of_files = filter(lambda f: os.path.isfile(os.path.join(motr_trace_dir, f)),
-                            os.listdir(motr_trace_dir))
-            # sort the files based on last modification time
-            list_of_files = sorted(list_of_files,
-                            key = lambda f: os.path.getmtime(os.path.join(motr_trace_dir, f)),
-                            reverse=True)
-            list_of_files = list_of_files[0:5]
-            for file in list_of_files:
+            latest_files = RGWSupportBundle._get_latest_files(directory=motr_trace_dir)
+            Log.info('Bundle motr trace files')
+            for file in latest_files:
                 infile = os.path.join(motr_trace_dir, file)
-                tmp_motr_trace_dir = os.path.join(RGWSupportBundle._tmp_src, 'motr_trace_files')
-                os.makedirs(tmp_motr_trace_dir, exist_ok=True)
                 outfile = os.path.join(tmp_motr_trace_dir, file)
-                # Convert m0trace file to human readable yaml format
-                SimpleProcess(f'm0trace -i {infile} -Y -o {outfile}.yaml').run()
-                SimpleProcess(f'xz {outfile}.yaml').run() # compress the output file
+                if size_limit:
+                    remaining_size_limit = RGWSupportBundle._get_remaining_folder_size(
+                        RGWSupportBundle._tmp_src, size_limit)
+                    if (os.path.getsize(infile) >= FilterLog._get_size_in_bytes(
+                            remaining_size_limit)):
+                        Log.info('size_limit is full.')
+                        break
+                    os.makedirs(tmp_motr_trace_dir, exist_ok=True)
+                    # Convert m0trace file to human readable yaml format
+                    SimpleProcess(f'm0trace -i {infile} -Y -o {outfile}.yaml').run()
+                    SimpleProcess(f'xz {outfile}.yaml').run()  # compress the output file
+                else:
+                    os.makedirs(tmp_motr_trace_dir, exist_ok=True)
+                    # Convert m0trace file to human readable yaml format
+                    SimpleProcess(f'm0trace -i {infile} -Y -o {outfile}.yaml').run()
+                    SimpleProcess(f'xz {outfile}.yaml').run()  # compress the output file
+
+        # copy addb logs
+        addb_log_path = os.path.join(log_dir, 'addb_files-*')
+        addb_dirs = glob.glob(addb_log_path)
+        for addb_dir in addb_dirs:
+            Log.info('Bundle addb files.')
+            if size_limit:
+                remaining_size_limit = RGWSupportBundle._get_remaining_folder_size(
+                    RGWSupportBundle._tmp_src, size_limit)
+                if (os.path.getsize(addb_dir) >= FilterLog._get_size_in_bytes(
+                        remaining_size_limit)):
+                    Log.info('size_limit is full.')
```

Comment:
Log.warning('Exhausted support bundle size limit, ADDB logs cant be collected further')

Reply:
done